Formula/g/gollama.rb

```diff
@@ -0,0 +1,23 @@
+class Gollama < Formula
+  desc "Go manage your Ollama models"
+  homepage "https://smcleod.net"
+  url "https://github.com/sammcj/gollama/archive/refs/tags/vv1.20.5.tar.gz"
+  sha256 "17335e526725cd7952f2ac2b3a9d7f5f556cc0aba24fa14e894e88cfba108773"
+  license "MIT"
+  head "https://github.com/sammcj/gollama.git", branch: "main"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", "-mod=readonly", *std_go_args(ldflags: "-s -w -X main.Version=#{version}", output: bin/"gollama")
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/gollama -v")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
@SMillerDev thank you for the detailed guidance - I have added an additional test, even though a simple one since it is a TUI app.